### ISSUE TITLE
Magnet file added to Transmission as paused even with "addpaused = no"

### ISSUE THREAD

**Issue by monchote** [Flexget/Flexget#672]
One of my flexget tasks uses `thetvdb_lookup` and `content_filename` to get the right path and filename for an Anime show. After the torrent magnetizes and the file gets renamed, the torrent stays as paused in Transmission. The other task I have doesn't use `thetvdb_lookup` or `content_filename` and starts downloading as soon as the torrent magnetizes.

Here is my `config.yml` for the problematic task:

```
templates:
  torrent:
    transmission:
      host: localhost
      port: 9091
      username: ******
      password: ******
      main_file_only: yes
      include_subs: yes
      magnetization_timeout: 90
      addpaused: no
  tv:
    template: torrent
    series:
      settings:
        all:
          quality: 720p+
          specials: no
tasks:
  sutorrents:
    template: tv
    thetvdb_lookup: yes
    rss:
      url: http://latorrent.com/anime/?rss=1
      link: torrent:magnetURI
    set:
      path: /media/samsung/Shows/{{tvdb_series_name}}/Season {{tvdb_season}}/
      content_filename: "{{ tvdb_series_name }} - {{ tvdb_season }}x{{ tvdb_episode|pad(2) }}"
    series:
      all:
        - Naruto Shippuuden:
            from_group: DeadFish
```

And here is what `flexget --debug execute` spits out for the task in question:

```
2015-12-03 13:12 DEBUG    series        sutorrents      adding release <SeriesParser(data=[DeadFish] Naruto Shippuuden - 440 [720p][AAC].mp4,name=Naruto Shippuuden,id=440,season=0,episode=440,quality=720p aac,proper=0,status=OK)> into episode
2015-12-03 13:12 DEBUG    series        sutorrents      -> added <Release(id=None,quality=720p aac,downloaded=None,proper_count=0,title=[DeadFish] Naruto Shippuuden - 440 [720p][AAC].mp4)>
2015-12-03 13:12 DEBUG    series        sutorrents      Naruto Shippuuden episode type totals: {u'sequence': 1}
2015-12-03 13:12 VERBOSE  series        sutorrents      identified by is currently on `auto` for Naruto Shippuuden. Multiple id types may be accepted until it locks in on the appropriate type.
2015-12-03 13:12 DEBUG    series        sutorrents      identified_by set to 'auto' based on series history
2015-12-03 13:12 DEBUG    series        sutorrents      start with episodes: [u'[DeadFish] Naruto Shippuuden - 440 [720p][AAC].mp4']
2015-12-03 13:12 DEBUG    series        sutorrents      current episodes: [u'[DeadFish] Naruto Shippuuden - 440 [720p][AAC].mp4']
2015-12-03 13:12 DEBUG    series        sutorrents      quality req: 720p+
2015-12-03 13:12 DEBUG    series        sutorrents      -------------------- process_propers -->
2015-12-03 13:12 DEBUG    series        sutorrents      propers - downloaded qualities: {}
2015-12-03 13:12 DEBUG    series        sutorrents      continuing w. episodes: [u'[DeadFish] Naruto Shippuuden - 440 [720p][AAC].mp4']
2015-12-03 13:12 DEBUG    series        sutorrents      best episode is: [DeadFish] Naruto Shippuuden - 440 [720p][AAC].mp4
2015-12-03 13:12 DEBUG    series        sutorrents      -------------------- episode tracking -->
2015-12-03 13:12 DEBUG    series        sutorrents      get_latest_release returning None, no downloaded episodes found for: Naruto Shippuuden
2015-12-03 13:12 DEBUG    series        sutorrents      latest download: None
2015-12-03 13:12 DEBUG    series        sutorrents      current: <Episode(id=124,identifier=440,season=0,number=440)>
2015-12-03 13:12 VERBOSE  task          sutorrents      ACCEPTED: `[DeadFish] Naruto Shippuuden - 440 [720p][AAC].mp4` by series plugin because matches quality
2015-12-03 13:12 DEBUG    urlrewriter   sutorrents      Checking 1 entries
2015-12-03 13:12 DEBUG    download      sutorrents      Accepting magnet url for [DeadFish] Naruto Shippuuden - 440 [720p][AAC].mp4
2015-12-03 13:12 VERBOSE  details       sutorrents      Summary - Accepted: 1 (Rejected: 0 Undecided: 24 Failed: 0)
2015-12-03 13:12 DEBUG    transmission  sutorrents      Successfully connected to transmission.
2015-12-03 13:12 DEBUG    api_tvdb      sutorrents      Looking up tvdb information for {u'tvdb_id': None, u'name': u'Naruto Shippuuden'}
2015-12-03 13:12 DEBUG    api_tvdb      sutorrents      Series <name=Naruto Shippuuden,tvdb_id=None> not found in cache, looking up from tvdb.
2015-12-03 13:12 DEBUG    requests      sutorrents      Fetching http://www.thetvdb.com/api/GetSeries.php?seriesname=Naruto%20Shippuuden&language=en
2015-12-03 13:12 DEBUG    requests      sutorrents      Fetching http://www.thetvdb.com/api/4D297D8CFDE0E105/mirrors.xml
2015-12-03 13:12 DEBUG    requests      sutorrents      Fetching http://thetvdb.com/api/4D297D8CFDE0E105/series/79824/en.xml
2015-12-03 13:12 DEBUG    api_tvdb      sutorrents      Looking up tvdb information for {u'tvdb_id': 79824, u'name': u'Naruto Shippuuden'}
2015-12-03 13:12 DEBUG    api_tvdb      sutorrents      Series <name=Naruto Shippuuden,tvdb_id=79824> information restored from cache.
2015-12-03 13:12 DEBUG    api_tvdb      sutorrents      Episode Naruto Shippuden.440 not found in cache, looking up from tvdb.
2015-12-03 13:12 DEBUG    requests      sutorrents      Fetching http://thetvdb.com/api/4D297D8CFDE0E105/series/79824/absolute/440/en.xml
2015-12-03 13:12 INFO     transmission  sutorrents      "[DeadFish] Naruto Shippuuden - 440 [720p][AAC].mp4" torrent added to transmission
2015-12-03 13:12 DEBUG    transmission  sutorrents      Waiting 90 seconds for "[DeadFish] Naruto Shippuuden - 440 [720p][AAC].mp4" to magnetize
2015-12-03 13:13 DEBUG    transmission  sutorrents      File [DeadFish] Naruto Shippuuden - 440 [720p][AAC].mp4 renamed to Naruto Shippuden - 17x79.mp4
2015-12-03 13:13 DEBUG    transmission  sutorrents      Downloading 1 of 1 files in torrent.
2015-12-03 13:13 DEBUG    series        sutorrents      on_task_learn
2015-12-03 13:13 DEBUG    series        sutorrents      marking 1 releases as downloaded for <Entry(title=[DeadFish] Naruto Shippuuden - 440 [720p][AAC].mp4,state=accepted)>
2015-12-03 13:13 DEBUG    seen          sutorrents      Learned '52BA27A861C974EDD5B96DCBB8E22BBBFA893318' (field: torrent_info_hash)
2015-12-03 13:13 DEBUG    seen          sutorrents      Learned '[DeadFish] Naruto Shippuuden - 440 [720p][AAC].mp4' (field: title)
2015-12-03 13:13 DEBUG    seen          sutorrents      Learned 'magnet:?xt=urn:btih:52BA27A861C974EDD5B96DCBB8E22BBBFA893318&dn=deadfish+naruto+shippuuden+440+720p+aac+mp4&tr=udp%3A%2F%2Ftracker.publicbt.com%2Fannounce&tr=udp%3A%2F%2Fglotorrents.pw%3A6969%2Fannounce' (field: url)
2015-12-03 13:13 DEBUG    util.simple_persistence sutorrents      Flushing simple persistence for task sutorrents to db.
2015-12-03 13:13 DEBUG    task_queue                    task queue shut down
2015-12-03 13:13 INFO     ipc.rpyc                      server has terminated
2015-12-03 13:13 INFO     ipc.rpyc                      listener closed
2015-12-03 13:13 DEBUG    util.simple_persistence                 Flushing simple persistence for task None to db.
2015-12-03 13:13 DEBUG    manager                       Shutting down
```

Is this a limitation or something I haven't setup properly?

**Comment by crawln45** [comment]
Nothing in that debug from what I can tell is wrong. So I'm not exactly sure what the issue is.

**Comment by monchote** [comment]
Anything I can do to help debugging the issue? I can reproduce it consistently.

**Comment by monchote** [comment]
I'm using the latest version of Flexget and transmissionrpc:

```
$ flexget -V
1.2.399
You are on the latest release.
$ pip show transmissionrpc
---
Metadata-Version: 1.1
Name: transmissionrpc
Version: 0.11
```

**Comment by crawln45** [comment]
I don't personally use transmission myself. I've looked through the code a few times but not familiar with it. Will check it out if I can.

**Comment by OmgOhnoes** [comment]
Couple of questions:

What is the default state for adding torrents to your client? Do they start automatically or are they added paused?

Second, you say that the torrent "stays" paused. By default Magnet links are added in an unpaused state because they have to magnetize before rename. So does the magnet add in an unpaused state THEN pause, or is it added in a paused state from the start?

**Comment by cvium** [comment]
Default is adding the torrents in the unpaused state. Magnet links are never paused.

**Comment by monchote** [comment]
I'm running Flexget and Transmission on a Raspberry Pi 2.

The Magnet link gets added unpaused. As soon as it magnetizes and gets renamed, it also gets paused.

The other task I have (which doesn't rename) has never displayed this problem.

**Comment by tarzasai** [comment]
this is my test:
```
tasks:
  testt1:
    disable:
      - seen
      - seen_info_hash
    thetvdb_lookup: yes
    mock:
      - { title: 'The.Big.Bang.Theory.S09E11.HDTV.x264-LOL[ettv]', url: 'magnet:?xt=urn:btih:DCC70364B791CB6CEBAADA04C4093C220D8257EE&dn=The+Big+Bang+Theory+S09E11+HDTV+x264-LOL%5Bettv%5D&tr=udp%3A%2F%2Ftracker.glotorrents.com%3A6969%2Fannounce&tr=udp%3A%2F%2Fglotorrents.pw%3A6969%2Fannounce&tr=http%3A%2F%2Ftracker.trackerfix.com%2Fannounce&tr=udp%3A%2F%2Feddie4.nl%3A6969%2Fannounce&tr=udp%3A%2F%2F9.rarbg.to%3A2710%2Fannounce&tr=udp%3A%2F%2F9.rarbg.me%3A2710%2Fannounce&tr=udp%3A%2F%2Ftracker.aletorrenty.pl%3A2710%2Fannounce&tr=udp%3A%2F%2Ftracker.opentrackr.org%3A1337%2Fannounce&tr=http%3A%2F%2Ftracker.tfile.me%2Fannounce&tr=http%3A%2F%2Fexplodie.org%3A6969%2Fannounce&tr=udp%3A%2F%2Fp4p.arenabg.com%3A1337&tr=http%3A%2F%2Fbigfoot1942.sektori.org%3A6969%2Fannounce&tr=udp%3A%2F%2Fopen.demonii.com%3A1337%2Fannounce&tr=udp%3A%2F%2Ftracker.openbittorrent.com%3A80%2Fannounce&tr=udp%3A%2F%2Ftracker.trackerfix.com%3A80%2Fannounce&tr=udp%3A%2F%2Ftracker.coppersurfer.tk%3A6969%2Fannounce' }
    all_series:
      tracking: backfill
    accept_all: yes
    transmission:
      host: 192.168.1.200
      port: 9092
      addpaused: no
      main_file_only: yes
      include_subs: yes
      magnetization_timeout: 90
      path: C:\Storage\Videos\Stage\series
      content_filename: "{{ tvdb_series_name }} - {{ tvdb_season }}x{{ tvdb_episode|pad(2) }}"
```
output:
```
2015-12-27 13:40 VERBOSE  task_queue                    There are 1 tasks to execute. Shutdown will commence when they have completed.
2015-12-27 13:40 VERBOSE  details       testt1          Produced 1 entries.
2015-12-27 13:40 VERBOSE  task          testt1          ACCEPTED: `The.Big.Bang.Theory.S09E11.HDTV.x264-LOL[ettv]` by series plugin because choosing best available quality
2015-12-27 13:40 VERBOSE  details       testt1          Summary - Accepted: 1 (Rejected: 0 Undecided: 0 Failed: 0)
2015-12-27 13:40 INFO     api_tvdb      testt1          Setting all series to expire
2015-12-27 13:40 VERBOSE  api_tvdb      testt1          Data for The Big Bang Theory has expired, refreshing from tvdb
2015-12-27 13:40 INFO     transmission  testt1          "The.Big.Bang.Theory.S09E11.HDTV.x264-LOL[ettv]" torrent added to transmission
```
the torrent is added and downloaded immediatley and stay **un**paused, as intended behaviour.
oh and the file gets renamed too, `The Big Bang Theory - 9x11.mp4.part`.
can you try your config on a different machine?
(transmission v2.84 on Windows 7 here)

**Comment by monchote** [comment]
Hi there, I added a 3 second sleep right before starting the torrent in `plugin_transmission.py` and the problem has completely disappeared.

As I mentioned, I'm running Flexget and Transmission on a Raspberry Pi 2 and the target disk is an external USB drive which is in power-save mode most of the time.

My hypothesis is that `cli.start_torrent(r.id)` gets called before the USB drive has waken up and the file has been renamed. Does it make sense at all? If so, why would the renaming operation not be synchronous? 

The following is the patch I'm running. I understand this is not the correct solution but I'd appreciate your feedback before I create a proper PR.

```python
diff --git a/flexget/plugins/plugin_transmission.py b/flexget/plugins/plugin_transmission.py
index efd81dc..f30e378 100644
--- a/flexget/plugins/plugin_transmission.py
+++ b/flexget/plugins/plugin_transmission.py
@@ -588,6 +588,8 @@ class PluginTransmission(TransmissionBase):
                 # prevents downloading data before we set what files we want
                 if ('paused' in options['post'] and options['post']['paused'] == False or
                    'paused' not in options['post'] and cli.get_session().start_added_torrents == True):
+                        from time import sleep
+                        sleep(3)
                         cli.start_torrent(r.id)
 
             except TransmissionError as e:
```

Regards

**Comment by tarzasai** [comment]
In that form it would be a mandatory pause for everybody - only needed by you for now.
I think you can add something in your task to wake up your drive instead, for example the `exec` plugin - perhaps on_output/for_accepted - to do something like write a dummy file on that drive.
and the `sleep` plugin could help too.

**Comment by rarguello-twitter** [comment]
I've been trying the `exec` plugin writing to a dummy file on `for_accepted` but I suspect that gets executed after the magnet link gets added so it doesn't help.

I agree my test patch would add delay even when not needed (and so I stated in my previous comment). However I think this race condition should not happen in any hardware configuration and that's why I was asking for some leads from you guys.

[`rename_torrent_path `](https://pythonhosted.org/transmissionrpc/reference/transmissionrpc.html#transmissionrpc.Client.rename_torrent_path) has a timeout param which defaults to `None`. Am I right in assuming that would always make it wait for renaming operation to finish?

**Comment by monchote** [comment]
I've been trying the `exec` plugin writing to a dummy file on `for_accepted` but I suspect that gets executed after the magnet link gets added so it doesn't help.

I agree my test patch would add delay even when not needed (and so I stated in my previous comment). However I think this race condition should not happen in any hardware configuration and that's why I was asking for some leads from you guys.

[`rename_torrent_path`](https://pythonhosted.org/transmissionrpc/reference/transmissionrpc.html#transmissionrpc.Client.rename_torrent_path) has a timeout param which defaults to `None`. Am I right in assuming that would always make it wait for renaming operation to finish?

**Comment by tarzasai** [comment]
> I've been trying the exec plugin writing to a dummy file on for_accepted but I suspect that gets executed after the magnet link gets added so it doesn't help.

and what about on_start / on_input / on_filter?

> rename_torrent_path has a timeout param which defaults to None. Am I right in assuming that would always make it wait for renaming operation to finish?

I don't think so, looks like they just pass it to the http_handler object in client.py.

**Comment by monchote** [comment]
> and what about on_start / on_input / on_filter?

That would wake the hard drive up every single time flexget gets executed, which I'd like to avoid.

**Comment by tarzasai** [comment]
> That would wake the hard drive up every single time flexget gets executed

I think you can try something like:
```
on_filter:
  for_accepted: if {{dunno}} do whatever
```
it should run before the output phase.

**Comment by tarzasai** [comment]
fixed along #705

**Comment by monchote** [comment]
This issue didn't get fixed with the patch for #705 as the issue is exactly the opposite.

**Comment by tarzasai** [comment]
not for me, I've tried again this [test](https://github.com/Flexget/Flexget/issues/672#issuecomment-167406271) and everything works as intended.
